pkg/controller/cronjob/utils.go

```diff
@@ -266,3 +282,35 @@ func (o byJobStartTime) Less(i, j int) bool {
 
 	return (*o[i].Status.StartTime).Before(*o[j].Status.StartTime)
 }
+
+// adoptJobs applies missing ControllerRefs to Jobs created by a CronJob.
+//
+// This should only happen if the Jobs were created by an older version of the
+// CronJob controller, since from now on we add ControllerRef upon creation.
+//
+// CronJob doesn't do actual adoption because it doesn't use label selectors to
+// find its Jobs. However, we should apply ControllerRef for potential
+// server-side cascading deletion, and to advise other controllers we own these
+// objects.
+func adoptJobs(sj *batchv2alpha1.CronJob, js []batchv1.Job, jc jobControlInterface) error {
+	var errs []error
+	sjControllerRef := *newControllerRef(sj)
+	for i := range js {
+		job := &js[i]
+		controllerRef := controller.GetControllerOf(job)
+		if controllerRef != nil {
+			continue
+		}
+		job.OwnerReferences = append(job.OwnerReferences, sjControllerRef)
+		updatedJob, err := jc.UpdateJob(job.Namespace, job)
```

Comment:
> What is the advantage of Patch? I used to think it would retry ResourceVersion conflicts for you on the server-side, but I recently found out it doesn't. 

This is surprising. I've been told that Patch is the best practice and apiserver will retry for you. Is it not the case? @caesarxuchao @lavalamp